fix(yield-protocol): update to use latest strategies

marcomariscal opened the pull request:
## Description

Update Yield Protocol strategies to the latest (new strategies launched).

## Checklist

- [x] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] (optional) As a contributor, my Ethereum address/ENS is:
- [ ] (optional) As a contributor, my Twitter handle is:

## How to test?

`0xf8cc4a0502145824a9b27a23dd9211c9c84dcfec` should have a `YSUSDC6MJD` (Yield Protocol June/December USDC Strategy) balance.

immasandwich commented:
@marcomariscal I refactored Yield Protocol according to some new patterns that we have in our system (templates). This helps us with optimizations that we have coming REALLY soon!

Since there would be huge conflicts, I went ahead and updated the strategies here: https://github.com/Zapper-fi/studio/commit/d34c1db2c1ccdf50ae3a61e3005a57830a46f172

Thanks!

marcomariscal commented:
> @marcomariscal I refactored Yield Protocol according to some new patterns that we have in our system (templates). This helps us with optimizations that we have coming REALLY soon!
> 
> Since there would be huge conflicts, I went ahead and updated the strategies here: [d34c1db](https://github.com/Zapper-fi/studio/commit/d34c1db2c1ccdf50ae3a61e3005a57830a46f172)
> 
> Thanks!

Awesome thanks so much @immasandwich!
